packages/toc/src/utils/markdown.ts

```diff
@@ -253,34 +253,36 @@ function parseHeading(line: string, nextLine?: string): IHeader | null {
   let match = line.match(/^([#]{1,6}) (.*)/);
   if (match) {
     return {
-      text: cleanTitle(match[2]),
+      text: removeHTMLTags(cleanTitle(match[2])),
       level: match[1].length,
       raw: line,
       skip: skipHeading.test(match[0])
     };
   }
+
```

Comment:
I have moved this condition to handle the below scenario:

<h1>Hello World</h1>
===

The <h1> tag should be addressed first and then the ===.